erpnext/stock/doctype/item/item.py

```diff
@@ -33,6 +33,11 @@ def autoname(self):
 		if frappe.db.get_default("item_naming_by")=="Naming Series" and not self.variant_of:
 			from frappe.model.naming import make_autoname
 			self.item_code = make_autoname(self.naming_series+'.#####')
+		elif frappe.db.get_default("item_naming_by")=="Naming Series" and self.variant_of:
+			item_code_suffix = ""
+			for attribute in self.attributes:
+				item_code_suffix += "-" + str(attribute.attribute_value)
```

Comment:
I was thinking about abbreviations but they don't exist in the Numeric Value attributes so I opted for using just the values because the attribute name could be really long.